File: Android/app/src/main/java/com/pokeapi/lpiem/pokeapiandroid/Provider/SocialNetworks/GoogleApiProvider.java
package com.pokeapi.lpiem.pokeapiandroid.Provider.SocialNetworks;

import com.google.android.gms.auth.api.signin.GoogleSignIn;
import com.google.android.gms.auth.api.signin.GoogleSignInClient;
import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
import com.pokeapi.lpiem.pokeapiandroid.View.MainActivity;

public class GoogleApiProvider implements GoogleApiInterface {
    private GoogleSignInClient googleSignInClient;

    @Override
    public void logOut() {

    }

    @Override
    public Object getData() {
        return null;
    }

    public void initGoogleApi(){
        googleSignInClient = MainActivity.initGoogleSignInApi();
    }

    public GoogleSignInClient getGoogleSignInClient(){
        return this.googleSignInClient;
    }
}

File: Android/app/src/main/java/com/pokeapi/lpiem/pokeapiandroid/Provider/SocialNetworks/GoogleApiProvider.kt
package com.pokeapi.lpiem.pokeapiandroid.Provider.SocialNetworks

import com.google.android.gms.auth.api.signin.GoogleSignInClient
import com.pokeapi.lpiem.pokeapiandroid.View.MainActivity

class GoogleApiProvider : GoogleApiInterface {
    override fun logIn() {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    var googleSignInClient: GoogleSignInClient? = null

    override fun logOut() {

    }

    fun initGoogleApi() {
        googleSignInClient = MainActivity.initGoogleSignInApi()
    }
}
